source/tit/geom/grid.hpp

```diff
@@ -31,103 +33,124 @@ class Grid final {
 public:
 
   /// Point type.
-  using Point = std::ranges::range_value_t<Points>;
+  using Vec = std::ranges::range_value_t<Points>;
 
-  /// Bounding box type.
-  using PointBBox = BBox<Point>;
-
-  /// Numeric type used by the point type.
-  using Real = vec_num_t<Point>;
-
-  /// Numeric type used by the point type.
-  static constexpr auto Dim = vec_dim_v<Point>;
-
-private:
-
-  Points points_;
-  PointBBox grid_bbox_;
-  Vec<size_t, Dim> num_cells_;
-  Point cell_size_;
-  Multivector<size_t> cell_points_;
-
-public:
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
   /// Initialize and build the multidimensional grid.
   ///
-  /// @param spacing Grid cell size, typically 2x of the particle spacing.
-  constexpr explicit Grid(Points points, Real spacing)
+  /// @param size_hint Grid cell size hint, typically 2x of the particle
+  ///                  spacing.
+  constexpr explicit Grid(Points points, vec_num_t<Vec> size_hint)
       : points_{std::move(points)} {
     TIT_PROFILE_SECTION("Grid::Grid()");
-    TIT_ASSERT(spacing > 0.0, "Spacing must be positive.");
+    TIT_ASSERT(size_hint > 0.0, "Cell size hint must be positive!");
     if (std::ranges::empty(points_)) return;
```

Comment:
This check must be removed, since all the data structures would not be built, and we'll need to check for that also in the search function.